gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -131,30 +132,24 @@ def __repr__(self):
                                                  self.alts,
                                                  self.formats,
                                                  self.contigs]])
-
-  def _values_asdict(self, header):
-    """Converts PyVCF header values to ordered dictionaries."""
-    ordered_dict = collections.OrderedDict()
-    for key in header:
-      # These methods were not designed to be protected. They start with an
-      # underscore to avoid conflicts with field names. For more info, see
-      # https://docs.python.org/2/library/collections.html#collections.namedtuple
-      ordered_dict[key] = header[key]._asdict()  # pylint: disable=W0212
-    return ordered_dict
-
   def _get_infos_pysam(self, infos):
+    self.verify_infos(infos)
     results = collections.OrderedDict()
     for item in infos.items():
       result = collections.OrderedDict()
       result['id'] = item[0]
-      result['num'] = item[1].number if item[1].number != '.' else None
-      result['type'] = item[1].type
+      result['num'] = item[1].number
+      result['type'] = item[1].record['Type']
       result['desc'] = item[1].description
       # Pysam doesn't return these fields in info
-      result['source'] = None
-      result['version'] = None
+      result['source'] = (item[1].record['Source'].strip("\"")
+                          if 'Source' in item[1].record and
+                          item[1].record['Source'] is not None else None)
+      result['version'] = (item[1].record['Version'].strip("\"")
+                           if 'Version' in item[1].record and
+                           item[1].record['Version'] is not None else None)
       results[item[0]] = result
-    return dict(results.items())
+    return results
 
   def _get_filters_pysam(self, filters):
     results = collections.OrderedDict()
```

Comment:
Similar to `_get_info_pysam()` let's assign `item[0]` to a meaningful variable. 
Also we could check the 'PASS' in this look and skip those items instead of deleting them later"
```
id = item[0]
if id == 'PASS'
  continue
result['id'] = id
...
results[id] = result
```